.gitignore

```diff
@@ -11,5 +11,11 @@ packages
 *.js.deps
 *.js.map
 
+# Include gitignore
+!.gitignore
+
+# Include .test_config
+!.test_config
+
 # Include when developing application packages.
-pubspec.lock
\ No newline at end of file
+pubspec.lock
```

Comment:
Habit born of safety and convenience really.   Typically including hidden files is an exception and, because they're hidden, they can slip through the cracks on commits. From a development perspective, it's a convenient way to be environment agnostic.  I use WebStorm, which wants to toss an .idea folder in the project root.  Having git ignore hidden files/directories reduces the need  to manage .gitignore per environment and remember not to accidentally propagate those customizations during a commit.  Is there a better way to handle this? 

Regardless, I'll pull the change.